### PR TITLE
Reintroduced meters on strength page using device-specific txPower value (or default)

### PR DESCRIPTION
**Pull request by Chippelius**

### CODE REVIEW

**Comment by Chippelius** [comment]
This branch was branched form fix-devices-crash. Merge that branch before this one to see the actual changes this pull request contains.

File: Implementation/FindMyBLEDevice/FindMyBLEDevice/Services/Bluetooth/Bluetooth.cs [hunk 1]
```diff
@@ -91,12 +91,35 @@ public void StartRssiPolling(String btguid, Action<int> updateRssi, Action conne
 
                         if (!(connected is null)) connected.Invoke();
 
+                        int txPower = Constants.TxPowerDefault;
+                        try
+                        {
+                            var service = await device.GetServiceAsync(Guid.ParseExact("00001804-0000-1000-8000-00805f9b34fb", "d"), token);
```
Comment (Wandinad) [inline Implementation/FindMyBLEDevice/FindMyBLEDevice/Services/Bluetooth/Bluetooth.cs:97]: What are the static Guids in this line and in line 100?

Comment (Chippelius) [reply Implementation/FindMyBLEDevice/FindMyBLEDevice/Services/Bluetooth/Bluetooth.cs:97]: [This formula](https://github.com/amosproj/amos2022ss05-find-my-hearing-aid/blob/2cbe55a03aaa3f5f8507f46d60da0880ee205a20/Implementation/FindMyBLEDevice/FindMyBLEDevice/ViewModels/StrengthViewModel.cs#L157), which converts RSSI values to meters, relies (next to the rssi itself) on two factors, the measured power and the environmental factor. While the environmental factor can only be guessed, the measured power (also called txPower, short for transmitted power) may be provided by the device whose rssi is being measured. 
If this txPower value is being provided by the device, then that is via the [txPower Service](https://github.com/xabre/xamarin-bluetooth-le/blob/9a8c22e0980ce43a17aa662c3f175b179b8291c9/Source/Plugin.BLE.Abstractions/KnownServices.cs#L45), or rather the [Tx Power Level Characteristic](https://github.com/xabre/xamarin-bluetooth-le/blob/9a8c22e0980ce43a17aa662c3f175b179b8291c9/Source/Plugin.BLE.Abstractions/KnownCharacteristics.cs#L106) within that service. 
The txPower value might be provided by the device [through other means as well](https://stackoverflow.com/a/24596727), but extracting it from those different ways would require device-specific knowledge. 

Note that our arduino test devices with the programming I provided do not offer their txPower value through this service. I did, however, extended the programming of my arduino to provide this service and characteristic and successfully tested it with the app using different values for txPower.

Comment (Wandinad) [reply Implementation/FindMyBLEDevice/FindMyBLEDevice/Services/Bluetooth/Bluetooth.cs:97]: Alright, that makes sense. Thanks for the explanation!

**Review by Wandinad — APPROVED** [review]

**Pull request merged**